src/scripts/main.js

```diff
@@ -1,3 +1,180 @@
+  } else {
+    element.setAttribute('direction', 'DESC');
+  }
+
+  tbodyRows.sort((a, b) => {
+    let aContent = a.cells[index].textContent;
+    let bContent = b.cells[index].textContent;
+
+    if (element.getAttribute('direction') === 'DESC') {
+      aContent = b.cells[index].textContent;
+      bContent = a.cells[index].textContent;
+    }
+
+    switch (index) {
+      case 0:
+      case 1:
+        return aContent.localeCompare(bContent);
+      case 2:
+        return aContent.localeCompare(bContent);
+      case 3:
+        return aContent - bContent;
+      case 4:
+        const toNumber = (item) => item.slice(1).split(',').join('');
+
+        return toNumber(aContent) - toNumber(bContent);
+      default:
+    }
+  });
+  tbodyRows.forEach(item => tbody.append(item));
+});
+
+tbody.addEventListener('click', (e) => {
+  const hasActive = tbody.getElementsByClassName('active')[0];
+  const selected = e.target.closest('tr');
+
+  if (hasActive) {
+    hasActive.classList.remove('active');
+  }
+  selected.classList.add('active');
+});
+
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+const formField = [
+  {
+    name: 'name',
+    type: 'text',
+  },
+  {
+    name: 'position',
+    type: 'text',
+  },
+  {
+    name: 'age',
+    type: 'number',
+  },
+  {
+    name: 'salary',
+    type: 'number',
+  },
+];
+const selectValues = ['Tokyo', 'Singapore',
+  'London', 'New York', 'Edinburgh', 'San Francisco'];
+
+form.innerHTML = formField.map(field => `
+  <label>
+    ${field.name[0].toUpperCase() + field.name.slice(1)}:
+      <input
+        name="${field.name}"
+        type="${field.type}"
+        data-qa="${field.name}"
+      >
+  </label>
+`).join('');
+
+form.querySelector('[name="age"]').setAttribute('min', 18);
+form.querySelector('[name="age"]').setAttribute('max', 90);
+
+form.children[1].insertAdjacentHTML('afterend', `
+  <label>Office:
+    <select name="office" data-qa="office" selected>
+    <option value="">
+      ${selectValues.map(value => `
+      <option value="${value}">
+        ${value}
+      </option>`).join('')}
+    </select>
+  </label>
+`);
+
+const submit = document.createElement('button');
+
+submit.type = 'submit';
+submit.textContent = 'Save to table';
+
+form.appendChild(submit);
+document.body.appendChild(form);
+
+const submitButton = document.querySelector('button');
+
+const pushNotification = (title, type) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add(`notification`, type);
+  notification.setAttribute('data-qa', 'notification');
+
+  const notificationTitle = document.createElement('h2');
+
+  notificationTitle.className = 'title';
+  notificationTitle.innerText = title;
+  notification.appendChild(notificationTitle);
+  document.querySelector('body').appendChild(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 2000);
+};
+
+function dataValidation() {
```

Comment:
Consider renaming the function 'dataValidation' to 'validateData' for better readability and to follow the verb-noun naming convention for functions.